Add PrismObjectType tag to S3 objects

shimpeko opened the pull request:
What:
This changes add PrismObjectType tag for each object type, live/delayed/merged to S3 objects.

Why:
To apply S3 lifecycle rules that targets each types.

For example, I'm planning to move merged objects to Amazon S3 Intelligent-Tiering storage class using lifecycle rule.

shimpeko committed:
Add PrismObjectType tag to S3 objects

What:
This changes add PrismObjectType tag for each object type,
live/delayed/merged to S3 objects.

Why:
To apply S3 lifecycle rules that targets each types.

For example, I'm planning to move merged objects to Amazon S3
Intelligent-Tiering storage class using lifecycle rule.

eagletmt commented:
Could you split this pull request into two parts, one for the build fixes and one for the tagging, so that we can easily revert the changes separately?

shimpeko commented:
https://github.com/cookpad/prism/pull/9

shimpeko committed:
refactor: Use enum for object types

shimpeko committed:
refactor: simpler

shimpeko committed:
remove unused

shimpeko committed:
Add test for putLiveObjectFile with tagging

Test newly added createPutRequestWithTag() method through the
putLiveObjectFile().

Updated build.gradle to resolve log4j delegation loop.
ref. https://www.slf4j.org/codes.html#log4jDelegationLoop

Why the delegatoin loop started to happen? Because I have added test
case for method in a class which has @Slf4j annotation. I'm not sure why
the error doesn't occur in running application.

shimpeko commented:
@eagletmt could you take a look?

shimpeko committed:
no need to define variable

shimpeko commented:
thank you for the review
